src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,35 @@
+package seedu.addressbook.data.person;
+/**
+ * Provides basic methods and attributes for the Email Class, Address Class and Phone Class
+ * Contains two attributes, value (String of Address, Email or Phone) and the boolean isPrivate
+ * Contains methods for toString, hashCode, getting the private attribute isPrivate and 
+ */
+public class Contact {
+    private final String value;
+    private boolean isPrivate;
+    
+
+    public Contact(boolean isPrivate, String value) {
+        this.isPrivate = isPrivate;
+        this.value = value;
+    }
+
+    @Override
+    public String toString() {
+        return getValue();
+    }
+    
+    @Override
+    public int hashCode() {
+        return getValue().hashCode();
+    }
+    
+    public boolean isPrivate() {
```

Comment:
Do note that trivial public methods still needs to be commented.
Refer back to the coding standards.